src/system/documents/item.ts

```diff
@@ -1051,18 +1049,83 @@ export class CosmereItem<
 
     /* --- Helpers --- */
 
-    protected async getEnrichedDescription(): Promise<string | undefined> {
-        if (!this.hasDescription()) return;
-        if (
-            !(this as CosmereItem<DescriptionItemData>).system.description
-                ?.value
-        )
-            return;
-
-        return await TextEditor.enrichHTML(
-            (this as CosmereItem<DescriptionItemData>).system.description!
-                .value!,
+    protected async getDescriptionHTML(): Promise<string | undefined> {
+        if (!this.hasDescription()) return undefined;
+
+        const descriptionData = (this as CosmereItem<DescriptionItemData>)
+            .system.description?.value;
+        const description = await TextEditor.enrichHTML(descriptionData ?? '');
+
+        const traitsNormal = [];
+        const traitsExpert = [];
+        const traits = [];
+        if (this.system.traits) {
+            for (const [key, value] of Object.entries(
+                this.system.traits as TraitsItemData,
+            )) {
+                // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+                if (!value?.active) continue;
+
+                const traitLoc =
+                    CONFIG.COSMERE.traits.weaponTraits[key as WeaponTraitId] ??
+                    CONFIG.COSMERE.traits.armorTraits[key as ArmorTraitId];
+                let label = game.i18n!.localize(traitLoc.label);
+
+                // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+                if (value.expertise?.toggleActive) {
+                    label = `<strong>${label}</strong>`;
+                    traitsExpert.push(label);
+                } else {
+                    traitsNormal.push(label);
+                }
+            }
+
+            traits.push(...traitsNormal.sort(), ...traitsExpert.sort());
+        }
+
+        let action;
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+        if (this.system.activation?.cost?.value !== undefined) {
```

Comment:
Add a `this.hasActivation` call here to get the typing. You'll still need to check that the cost actually exists though.